### PR TITLE
Use 4.14.0 opam switch for building (includes upgrading ocamlformat to 0.24.1)

### PR DESCRIPTION
**Pull request by mshinwell**
This PR changes the build opam switch requirement to 4.14.0.  This allows use of the latest tooling.  It includes an upgrade of ocamlformat to 0.24.1.

There are a few minor changes in formatting.  The default value of a formatting option changed, so I've put the previous value in the file, saving a much larger change.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -86,30 +86,30 @@ jobs:
...
 
-    - name: Cache OCaml 4.12 and dune
+    - name: Cache OCaml 4.14 and dune
       uses: actions/cache@v1
       id: cache
       with:
-        path: ${{ github.workspace }}/ocaml-412/_install
-        key: ${{ matrix.os }}-cache-ocaml-412-dune-341
+        path: ${{ github.workspace }}/ocaml-414/_install
+        key: ${{ matrix.os }}-cache-ocaml-414
```
Comment (xclerc) [inline .github/workflows/build.yml:94]: (If the suffix is removed, possibly unpin dune.)

**Review by xclerc — COMMENTED** [review]
I think you may also want to update `.vscode/settings.json`,
`.vscode/tasks.json`, and `configure.ac`; not sure about
`ocaml/HACKING.jst.adoc`, which I would assume would be
updated when synchronizing the repositories.

**Comment by xclerc** [comment]
(Maybe also consider `flambda_backend.opam`
and `external/memtrace/memtrace.opam`.)

File: .github/workflows/build.yml (hunk 1 above)
Comment (mshinwell) [reply .github/workflows/build.yml:94]: Ah this was actually pinned.  Have updated to 3.6.1 which is what I currently get in a 4.14.0 switch by default.

**Comment by mshinwell** [comment]
> (Maybe also consider `flambda_backend.opam` and `external/memtrace/memtrace.opam`.)

I don't think we need to do these yet, they are already out of date and probably require other work.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -86,45 +86,45 @@ jobs:
...
         path: 'dune'
 
     - name: Build dune
       working-directory: dune
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
-        PATH=$GITHUB_WORKSPACE/ocaml-412/_install/bin:$PATH make release
-        cp dune.exe $GITHUB_WORKSPACE/ocaml-412/_install/bin/dune
+        PATH=$GITHUB_WORKSPACE/ocaml-414/_install/bin:$PATH make release
+        cp _boot/dune.exe $GITHUB_WORKSPACE/ocaml-414/_install/bin/dune
```
Comment (xclerc) [inline .github/workflows/build.yml:127]: (I am surprised it comes from `_boot`
rather than `_build`, but the checksums
are the same on my box.)

**Review by xclerc — APPROVED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (mshinwell) [reply .github/workflows/build.yml:127]: The wrapper script `dune.exe` (seems to be a recent addition?) runs the one in `_boot`, so I just copied that.

**Pull request merged**